gradcam.py

```diff
@@ -0,0 +1,268 @@
+        return output
+
+    @staticmethod
+    def backward(self, grad_output):
+        input, output = self.saved_tensors
+        grad_input = None
+
+        positive_mask_1 = (input > 0).type_as(grad_output)
+        positive_mask_2 = (grad_output > 0).type_as(grad_output)
+        grad_input = torch.addcmul(torch.zeros(input.size()).type_as(input),
+                                   torch.addcmul(torch.zeros(input.size()).type_as(input), grad_output,
+                                                 positive_mask_1), positive_mask_2)
+
+        return grad_input
+
+
+class GuidedBackpropReLUModel:
+    def __init__(self, model, use_cuda):
+        self.model = model
+        self.model.eval()
+        self.cuda = use_cuda
+        if self.cuda:
+            self.model = model.cuda()
+
+        def recursive_relu_apply(module_top):
+            for idx, module in module_top._modules.items():
+                recursive_relu_apply(module)
+                if module.__class__.__name__ == 'ReLU':
+                    module_top._modules[idx] = GuidedBackpropReLU.apply
+                
+        # replace ReLU with GuidedBackpropReLU
+        recursive_relu_apply(self.model)
+
+    def forward(self, input):
+        return self.model(input)
+
+    def __call__(self, input, index=None):
+        if self.cuda:
+            output = self.forward(input.cuda())
+        else:
+            output = self.forward(input)
+
+        if index == None:
+            index = np.argmax(output.cpu().data.numpy())
+
+        one_hot = np.zeros((1, output.size()[-1]), dtype=np.float32)
+        one_hot[0][index] = 1
+        one_hot = torch.from_numpy(one_hot).requires_grad_(True)
+        if self.cuda:
+            one_hot = torch.sum(one_hot.cuda() * output)
+        else:
+            one_hot = torch.sum(one_hot * output)
+
+        # self.model.features.zero_grad()
+        # self.model.classifier.zero_grad()
+        one_hot.backward(retain_graph=True)
+
+        output = input.grad.cpu().data.numpy()
+        output = output[0, :, :, :]
+
+        return output
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--use-cuda', action='store_true', default=False,
+                        help='Use NVIDIA GPU acceleration')
+    parser.add_argument('--image-path', type=str, default='./examples/both.png',
+                        help='Input image path')
+    args = parser.parse_args()
+    args.use_cuda = args.use_cuda and torch.cuda.is_available()
+    if args.use_cuda:
+        print("Using GPU for acceleration")
+    else:
+        print("Using CPU for computation")
+
+    return args
+
+def deprocess_image(img):
+    """ see https://github.com/jacobgil/keras-grad-cam/blob/master/grad-cam.py#L65 """
+    img = img - np.mean(img)
+    img = img / (np.std(img) + 1e-5)
+    img = img * 0.1
+    img = img + 0.5
+    img = np.clip(img, 0, 1)
+    return np.uint8(img*255)
+
+
+if __name__ == '__main__':
```

Comment:
Instead of having all of this code run in terminal, can you make it all into one big function? (i.e. w/model, feature_model, target_layer_names, use_cuda, image_path as args). Also, please no argparser ty